src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp

```diff
@@ -412,7 +412,113 @@ bool PDPServer2::process_change_acknowledgement(
     return true;
 }
 
-fastdds::rtps::ddb::DiscoveryDataBase& PDPServer2:: discovery_db()
+bool PDPServer2::process_disposals()
+{
+    EDPServer2* edp = static_cast<EDPServer2*>(mp_EDP);
+    fastrtps::rtps::WriterHistory* pubs_history = edp->publications_writer_.second;
+    fastrtps::rtps::WriterHistory* subs_history = edp->subscriptions_writer_.second;
+
+    // Get list of disposals from database
+    std::vector<fastrtps::rtps::CacheChange_t*> disposals = discovery_db_.changes_to_dispose();
+    // Iterate over disposals
+    for(auto change: disposals)
+    {
+        // Check that the change is actually a disposal
+        if (change->kind != fastrtps::rtps::ChangeKind_t::NOT_ALIVE_DISPOSED)
```

Comment:
This should be checked before, where the disposal list is populated

Reply:
This means just that the comprobation could be skipped. It is nothing wrong with it.

Reply:
:+1: